File: bale_3d_bot_backup/pages/api/payment.js
// pages/api/payment.js
export default async function handler(req, res) {
  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Method not allowed' });
  }

  const { chatId, amount, walletId, message } = req.body;

  // حالت اول: درخواست از دکمه ولت تست (ارسال walletId و message)
  if (walletId && message) {
    console.log(`✅ Simulated payment for wallet: ${walletId}`);
    return res.status(200).json({
      ok: true,
      transactionId: `TEST_${Date.now()}`,
      message: 'پرداخت آزمایشی با موفقیت انجام شد'
    });
  }

  // حالت دوم: درخواست از صفحه پرداخت واقعی (ارسال chatId و amount)
  if (chatId && typeof amount === 'number' && amount > 0) {
    const transactionId = `PAY_${Date.now()}_${Math.random().toString(36).substr(2, 6)}`;
    console.log(`✅ Payment: ${chatId} - ${amount} تومان`);
    return res.status(200).json({ ok: true, transactionId });
  }

  // در غیر این صورت خطا بده
  return res.status(400).json({ error: 'Invalid request: missing chatId+amount or walletId+message' });
}

File: bale_3d_bot_backup/pages/api/payment.py
import fetch from "node-fetch";

const BALE_BOT_TOKEN = process.env.BALE_BOT_TOKEN; // توکن را در فایل .env قرار بده

export default async function handler(req, res) {
  if (req.method !== "POST")
    return res.status(405).json({ error: "Method Not Allowed" });

  const { walletId, message } = req.body;
  if (!walletId || !message)
    return res.status(400).json({ error: "Missing walletId or message" });

  try {
    const response = await fetch(`https://core.bale.ai/api/v1/bot-message-send`, {
      method: "POST",
      headers: {
        "Authorization": `Bearer ${BALE_BOT_TOKEN}`,
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ to: walletId, type: "Text", text: message }),
    });

    if (!response.ok) {
      const errTxt = await response.text();
      return res.status(500).json({ error: `Send message failed: ${errTxt}` });
    }

    const data = await response.json();
    res.status(200).json(data);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
}
